FIX dump svmlight when data is read-only (#28111)
scikit-learn/scikit-learn: doc/whats_new/v1.4.rst

.. currentmodule:: sklearn

.. _changes_1_4_1:

Version 1.4.1
=============

**In Development**

Changelog
---------

:mod:`sklearn.datasets`
.......................

> - |Fix| :func:`datasets.dump_svmlight_file` now does not raise `ValueError` when `X`
>   is read-only, e.g., a `numpy.memmap` instance.

Reviewer: ```suggestion
- |Fix| :func:`datasets.dump_svmlight_file` now support read-only inputs for `X`
  such as `numpy.memmap` instances.
```
